### PR TITLE
smb/eve: convert to jsonbuilder

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3712

### CODE REVIEW

File: rust/src/smb/log.rs [hunk 1]
```diff
@@ -194,162 +195,164 @@ fn smb_common_header(state: &SMBState, tx: &SMBTransaction) -> Json
...
+            jsb.set_uint("size", x.size)?;
 
             let gs = fuid_to_string(&x.guid);
-            js.set_string("fuid", &gs);
+            jsb.set_string("fuid", &gs)?;
         },
         Some(SMBTransactionTypeData::NEGOTIATE(ref x)) => {
             if x.smb_ver == 1 {
-                let jsa = Json::array();
+                let mut jsa = JsonBuilder::new_array();
```
Comment (victorjulien) [inline rust/src/smb/log.rs:233]: @jasonish does this lead to a mem alloc? If so can we avoid it? Really all we need is to add a `"name":[ ...` and then later a `]`?

Comment (jasonish) [reply rust/src/smb/log.rs:233]: Yes it would.. Can use open_array here followed by a close.

Comment (inashivb) [reply rust/src/smb/log.rs:233]: Thanks, Jason!

**Comment by inashivb** [comment]
Replaced with #5068

**Pull request closed without merging**